src/security/core/src/dds_security_utils.c

```diff
@@ -0,0 +1,1058 @@
+            } else{
+                return CRYPTO_TRANSFORMATION_KIND_AES256_GCM;
+            }
+        default:
+            return CRYPTO_TRANSFORMATION_KIND_INVALID;
+    }
+}
+
+DDS_Security_CryptoTransformKind_Enum
+DDS_Security_protectionkind2transformationkind(
+     const DDS_Security_PropertySeq *properties,
+     DDS_Security_ProtectionKind protection)
+{
+    int keysize=256;
+    const DDS_Security_Property_t *key_size_property = NULL;
+    if( properties != NULL ){
+        key_size_property = DDS_Security_PropertySeq_find_property(
+                        properties, "dds.sec.crypto.keysize");
+        if (key_size_property != NULL) {
+            if (strcmp(key_size_property->value, "128") == 0) {
+                keysize = 128;
+            }
+        }
+    }
+
+    switch (protection) {
+        case DDS_SECURITY_PROTECTION_KIND_NONE:
+            return CRYPTO_TRANSFORMATION_KIND_NONE;
+        case DDS_SECURITY_PROTECTION_KIND_SIGN_WITH_ORIGIN_AUTHENTICATION:
+        case DDS_SECURITY_PROTECTION_KIND_SIGN:
+            if( keysize == 128 ){
+                return CRYPTO_TRANSFORMATION_KIND_AES128_GMAC;
+            } else{
+                return CRYPTO_TRANSFORMATION_KIND_AES256_GMAC;
+            }
+        case DDS_SECURITY_PROTECTION_KIND_ENCRYPT_WITH_ORIGIN_AUTHENTICATION:
+        case DDS_SECURITY_PROTECTION_KIND_ENCRYPT:
+            if( keysize == 128 ){
+                return CRYPTO_TRANSFORMATION_KIND_AES128_GCM;
+            } else {
+                return CRYPTO_TRANSFORMATION_KIND_AES256_GCM;
+            }
+        default:
+            return CRYPTO_TRANSFORMATION_KIND_INVALID;
+    }
+}
+
+/* for DEBUG purposes */
+void
+print_binary_debug(
+     char* name,
+     unsigned char *value,
+     uint32_t size)
+{
+    uint32_t i;
+    printf("%s: ",name );
+    for( i=0; i<  size; i++)
+    {
+        printf("%x",value[i]);
+    }
+    printf("\n");
+}
+
+void
+print_binary_properties_debug(
+     const DDS_Security_DataHolder *token)
+{
+    uint32_t i;
+    for (i = 0; i < token->binary_properties._length ; i++) {
+        print_binary_debug( token->binary_properties._buffer[i].name, token->binary_properties._buffer[i].value._buffer, token->binary_properties._buffer[i].value._length);
+    }
+
+}
+
+
+
+
+DDS_Security_config_item_prefix_t
+DDS_Security_get_conf_item_type(
+     const char *str,
+     char **data)
+{
+    DDS_Security_config_item_prefix_t kind = DDS_SECURITY_CONFIG_ITEM_PREFIX_UNKNOWN;
+    const char *CONFIG_FILE_PREFIX   = "file:";
+    const char *CONFIG_DATA_PREFIX   = "data:,";
+    const char *CONFIG_PKCS11_PREFIX = "pkcs11:";
+    size_t CONFIG_FILE_PREFIX_LEN   = strlen(CONFIG_FILE_PREFIX);
+    size_t CONFIG_DATA_PREFIX_LEN   = strlen(CONFIG_DATA_PREFIX);
+    size_t CONFIG_PKCS11_PREFIX_LEN = strlen(CONFIG_PKCS11_PREFIX);
+    char *ptr;
+
+    assert(str);
+    assert(data);
+    DDSRT_UNUSED_ARG(str);
+
+    ptr = NULL;//ddsrt_strchrs(str, " \t", false);
```

Comment:
NULL is used as pointer. Doesn't that break anything?
Can't strchrs() be moved from auth_utils to security_utils? Than it can be used in both locations.

Reply:
I moved strchrs to dds_security_utils in core component. So, we'll use ddssec_strchrs here, instead of NULL.

Reply:
Just a friendly reminder: https://github.com/eclipse-cyclonedds/cyclonedds/pull/177#issuecomment-490718462

I had expected ``strchrs`` to have been removed altogether ...